### ISSUE TITLE
Creating Progressive Web Applications 

### ISSUE THREAD

**Issue by seme1** [framework7io/framework7#1694]
This is a (multiple allowed):
* [] bug
* [X] enhancement
* [ ] feature-discussion (RFC)

It would be nice if Framework7 has a built-in/easy to implement way for building progressive web applications. Things such as instant load on launch, dynamic content loading of resources when needed an automatic content caching are all desired.

**Comment by seme1** [comment]
No feedback/updates on this ???

**Comment by valnub** [comment]
I don't think that F7 should provide PWA features since it's just a view layer framework.

PWA features like Service Workers and such are separate from F7.

Anyway, I'm just starting to dig into PWAs since Webkit support is coming. Maybe there's something that can put in.

**Comment by macdonst** [comment]
@seme1 the PhoneGap team has created a number of templates that use F7/Vue to create mobile apps that can run on Android, iOS or as a PWA. Check out the split view one here:

https://github.com/phonegap/phonegap-template-vue-f7-split-panel

We'd love some feedback.

**Comment by valnub** [comment]
@macdonst Is there an overview/list of all the templates that you guys created? I am still not sure what's available and what not.

**Comment by macdonst** [comment]
@valnub there is also:

https://github.com/phonegap/phonegap-template-vue-f7-blank
https://github.com/phonegap/phonegap-template-vue-f7-tabs

and @devgeeks is documenting creating an app using one from scratch.

**Comment by nolimits4web** [comment]
This issue was incorrectly closed on 2026-05-19 by a compromised credential (Shai-Hulud npm supply-chain worm). Reopening — apologies for the noise.
